### PR TITLE
Document CLI support for per interface sysctls

### PR DESCRIPTION
**Pull request by robmry**
**- What I did**

- https://github.com/moby/moby/pull/47686 adds driver-opt label `com.docker.network.endpoint.sysctls`, to support per-interface sysctls.

The extended `--network` syntax is needed, but wasn't documented.

The quoting needed to set more than one sysctl using `docker [create|run] --network driver-opt` or `docker network connect --driver-opt` isn't obvious - so, added tests and examples to the docs.

**- How I did it**

- Fix `TestNetworkConnectWithFlags` - because it didn't do anything, and is needed to test the new option.
- Document the v25.0 changes to allow more than one network connection in create/run, with the extended `--network` syntax.
- Added tests and examples to docs for per-interface sysctl setting.

**- How to verify it**

New tests added.

**- Description for the changelog**

```markdown changelog
Document the use of `driver-opt` in the CLI for setting per-interface sysctls.
```

### CODE REVIEW

File: docs/reference/commandline/container_run.md [hunk 1]
```diff
@@ -731,22 +748,38 @@ $ docker network create --subnet 192.0.2.0/24 my-net
...
-> **Note**
->
-> The default bridge network only allow containers to communicate with each other using
-> internal IP addresses. User-created bridge networks provide DNS resolution between
-> containers using container names.
+When connecting to more than one network, use the extended syntax for
+`--network` to supply options. Comma-separated options that can be
+specified in the extended `--network` syntax are `name`, which is mandatory,
+`alias`, `ip`, `ip6`, `mac-address`, `link-local-ip`, `sysctl` and
+`driver-opt`.
```
[the remark this answers is not in the document]
Comment (robmry) [reply docs/reference/commandline/container_run.md]: Done.

File: docs/reference/commandline/container_run.md [hunk 2]
```diff
@@ -731,22 +748,38 @@ $ docker network create --subnet 192.0.2.0/24 my-net
...
+is where you'll find them in `docker inspect` output. Network drivers may restrict
+the sysctl settings that can be modified and, to protect the operation of the
+network, new restrictions may be added in the future.
+
+```console
+$ docker network create --subnet 192.0.2.0/24 my-net
+$ docker run -itd --network=name=my-net,sysctl=ipv4.conf.log_martians=1,ip=192.0.2.42 busybox
+```
 
 For more information on connecting a container to a network when using the `run` command, see the ["*Docker network overview*"](https://docs.docker.com/network/).
```
[the remark this answers is not in the document]
Comment (robmry) [reply docs/reference/commandline/container_run.md]: Done.

**Review by dvdksn — APPROVED** [review]
LGTM

**Review by laurazard — APPROVED** [review]
LGTM

**Review by akerouanton — APPROVED** [review]

**Comment by robmry** [comment]
Updated to reflect a last-minute change of heart on the sysctl naming (`net.ipv6.conf.eth0.disable_ipv6` was shortened to `ipv6.conf.disable_ipv6`, now it's `net.ipv6.conf.IFNAME.disable_ipv6`).

**Review by akerouanton — APPROVED** [review]

**Review by dvdksn — APPROVED** [review]

**Review by rumpl — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**